lib/src/model.dart

```diff
@@ -4841,59 +4985,233 @@ abstract class LibraryContainer extends Nameable
+    _enums = [];
+    _exceptions = [];
+    _classes = [];
+    _constants = [];
+    _properties = [];
+    _functions = [];
+    _typedefs = [];
+  }
+
+  void addItem(Categorization c) {
+    if (_allItems.contains(c)) return;
+    _allItems.add(c);
+    if (c is Library) {
+      _libraries.add(c);
+    } else if (c is Enum) {
+      _enums.add(c);
+    } else if (c is Class) {
+      if (c.isErrorOrException) {
+        _exceptions.add(c);
+      } else {
+        _classes.add(c);
+      }
+    } else if (c is TopLevelVariable) {
+      if (c.isConst) {
+        _constants.add(c);
+      } else {
+        _properties.add(c);
+      }
+    } else if (c is ModelFunction) {
+      _functions.add(c);
+    } else if (c is Typedef) {
+      _typedefs.add(c);
+    } else {
+      throw UnimplementedError("Unrecognized element");
+    }
+  }
 
-  Category(this._name, this.package, this.config);
+  @override
+  // TODO(jcollins-g): make [Category] a [Warnable]?
+  Warnable get enclosingElement => null;
 
   @override
-  String get name => _name;
+  Element get element => null;
+
+  @override
+  String get name => categoryDefinition?.displayName ?? _name;
+
+  @override
+  String get sortKey => _name;
 
   @override
-  List<String> get containerOrder => config.categoryOrder;
+  List<String> get containerOrder => config.categories.categoryOrder;
 
   @override
   String get enclosingName => package.name;
 
+  @override
   PackageGraph get packageGraph => package.packageGraph;
+
+  @override
+  Library get canonicalLibrary => null;
+
+  @override
+  List<Locatable> get documentationFrom => [this];
+
+  @override
+  DocumentLocation get documentedWhere => package.documentedWhere;
+
+  bool _isDocumented;
+  @override
+  bool get isDocumented {
+    if (_isDocumented == null) {
+      _isDocumented = documentedWhere != DocumentLocation.missing &&
+          documentationFile != null;
+    }
+    return _isDocumented;
+  }
+
+  @override
+  String get fullyQualifiedName => name;
+
+  @override
+  String get href =>
+      isCanonical ? '${package.baseHref}apis/${name}-api.html' : null;
+
+  String get linkedName {
+    String unbrokenCategoryName = name.replaceAll(' ', '&nbsp;');
+    if (isDocumented) {
+      return '<a href="$href">$unbrokenCategoryName</a>';
+    } else {
+      return unbrokenCategoryName;
+    }
+  }
+
+  String _categoryNumberClass;
+
+  /// The position in the container order for this category.
+  String get categoryNumberClass {
+    if (_categoryNumberClass == null) {
+      _categoryNumberClass = "cp-${package.categories.indexOf(this)}";
+    }
+    return _categoryNumberClass;
+  }
+
+  /// Category name used in template as part of the class.
+  String get spanClass => name.split(' ').join('-').toLowerCase();
+
+  CategoryDefinition get categoryDefinition =>
+      config.categories.categoryDefinitions[sortKey];
+
+  @override
+  bool get isCanonical => categoryDefinition != null;
+
+  @override
+  String get kind => 'API';
```

Comment:
I think we should noodle on a slightly better name for this.

Reply:
todo added